### PR TITLE
Sprint 1, final PR

### PR DESCRIPTION
**Pull request by ink8bit**
# Sprint 1

## Build & run application

> The application will use default address `localhost:8080`

```console
cd cmd/shortener && go build && ./shortener
```

## Run tests

Go to the project root and run the following command:

```console
go test ./...
```

### CODE REVIEW

File: internal/shorten/shorten.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package shorten
+
+import (
+	"math/rand"
+	"strings"
+)
+
+// GenStr creates randomized string from letters and digits.
+func GenStr() string {
```
Comment (arttet) [inline internal/shorten/shorten.go:9]:
```suggestion
func GenRandomStr() string {
```

File: internal/app/handlers.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+	"urlshort/internal/config"
+	"urlshort/internal/storage"
+
+	"github.com/go-chi/chi/v5"
+)
+
+var (
+	// For testing purposes
+	findURL = storage.FindURL
```
Comment (arttet) [inline internal/app/handlers.go:17]: Please avoid global variables
If it is for testing purposes, it should be in the test package

File: internal/storage/storage.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
+package storage
+
+import (
+	"errors"
+	"urlshort/internal/shorten"
+)
+
+// NOTE: it's a temporary solution.
```
Comment (arttet) [inline internal/storage/storage.go:8]:
```suggestion
// TODO: it's a temporary solution.
```
I like this comment. But let's fix this issue and exclude using global variables.

File: internal/storage/storage.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+// shortUrls map structure: website_url -> short_url, i.e. https://go.dev -> http://localhost/abcdefg
+// origUrls map structure: short_url -> website_url, i.e. http://localhost/abcdefg -> https://go.dev
+// It gives us O(1) time complexity for searching original and shortened urls.
+var (
+	shortUrls = make(map[string]string)
+	origUrls  = make(map[string]string)
+)
+
+// SaveURL saves original and short urls to storage.
+func SaveURL(origURL string) string {
```
Comment (arttet) [inline internal/storage/storage.go:21]: What if we get an error?

**Review by arttet — CHANGES_REQUESTED** [review]
Please read it
https://cbea.ms/git-commit/
https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/

Please add linters in CI https://github.com/golangci/golangci-lint
Please add the code coverage report in CI https://about.codecov.io/blog/getting-started-with-code-coverage-for-golang/
Please add gofmt in CI https://github.com/marketplace/actions/check-code-formatting-using-gofmt

File: internal/storage/storage.go (hunk 4 above)
Comment (ink8bit) [reply internal/storage/storage.go]: I've change the function signature to `func SaveURL(origURL string) (string, error)`.

File: .github/workflows/gofmt.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+name: gofmt
+on:
+  push:
+    branches:
+      - master
+      - main
+  pull_request:
+
+jobs:
+  golangci:
```
Comment (arttet) [inline .github/workflows/gofmt.yml:10]:
```suggestion
  gofmt:
```

File: internal/app/router/router.go [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+	"urlshort/internal/app"
+
+	"github.com/go-chi/chi/v5"
+)
+
+func New(s *app.Server) http.Handler {
+	r := chi.NewRouter()
+
+	r.Get("/{id:^[0-9A-Za-z]+}", s.OriginURLHandler)
```
Comment (arttet) [inline internal/app/router/router.go:14]:
```suggestion
	r.Get("/{id:^[0-9A-Za-z]+$}", s.OriginURLHandler)
```

File: internal/config/flags.go [hunk 7]
```diff
@@ -0,0 +1,24 @@
+package config
+
+import (
+	"flag"
+	"os"
+)
+
+var (
+	Addr    string
```
Comment (arttet) [inline internal/config/flags.go:9]: Please don't use global variables

**Review by arttet — APPROVED** [review]

**Pull request merged**